File: src/listener.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

using namespace ros;

void chatterCallback(const std_msgs::String::ConstPtr& msg) {
    ROS_INFO("[Listener] I heard: [%s]\n", msg->data.c_str());
}

int main(int argc, char **argv) {

    // Initializes a new node named "listener_node"
    init(argc, argv, "listener_node");

    // Subscribes node to the topic chatter with a processing queue of 1000
    // and executes the callback function whenever a message is received
    NodeHandle node;
    Subscriber sub = node.subscribe("chatter", 1000, chatterCallback);

    // Enters a loop of listening for messages and executing callbacks
    spin();

    return 0;
}

File: src/scripts/listener.py
import rospy
from std_msgs.msg import String

def chatter_callback(message):
    rospy.loginfo(rospy.get_caller_id() + "I heard %s", message.data)

def listener():

    # Initializes a node with name listener and ensures unique name
    rospy.init_node('listener', anonymous=True)

    """ Creates a subscriber object to topic chatter expecting data 
    String and performs "chatter_callback" upon receiving a new message """
    rospy.Subscriber("chatter", String, chatter_callback)

    # Program goes into listening mode; executes callback w/ every message
    rospy.spin()

if __name__ == '__main__':
    listener()